### DISCUSSION TITLE
[Feature Request] Can we implement serde::Serialize for diesel::result::Error?

### DISCUSSION THREAD

**Discussion by jaseemabid** [diesel-rs/diesel#3174] [Ideas]
Title says it all.  Any reason to not do this? 

https://serde.rs/remote-derive.html is a workaround, but for a type as large as `diesel::result::Error`, it's a bit cumbersome.

**Comment by weiznich** [comment]
I would like to hear some better justification for this than "why not?".
Especially I would like to hear answers to the following questions:

* What use case do you have in mind? 
* How universal would useful would an Serialize implementation be for other people? Would they expect the same fields? Other fields?
* How exactly would you imaging implementing `Serialize` for `diesel::result::Error`, especially as it contains trait objects?
